### PR TITLE
Implement csrf-token in TRPC

### PR DESCRIPTION
**Pull request by williamputraintan**
Patch #200

### CODE REVIEW

File: application/backend/src/api/api-trpc-routes.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import { appRouter } from "../app-router";
+import { createContext } from "./routes/trpc-context";
+
+/**
+ * Defined for trpc routes
+ *
+ * @param fastify
+ */
+export const trpcRoutes = async (fastify: FastifyInstance) => {
+  fastify.addHook("preHandler", fastify.csrfProtection);
```
Comment (williamputraintan) [inline application/backend/src/api/api-trpc-routes.ts:12]: Thinking to make use of `fastify.addHook` for csrf-token checking, and so breaking this to routes before registering the `fastifyTRPCPlugin`.

**Comment by andrewpatto** [comment]
I guess I haven't thought too much about it - is it even possible for a CSRF exploit over TRPC?

**Comment by andrewpatto** [comment]
I mean I guess a lot of the "payload" for TRCP goes into the URL itself and therefore could be CSRF exploitable..

**Comment by williamputraintan** [comment]
I suppose yes, I guess it is just a more longer url for the exploitation ? aaand it is now equal regardless the internal api uses trpc or not both have csrf-token protection.

**Review by andrewpatto — APPROVED** [review]

**Pull request merged**